index.html

```diff
@@ -1147,59 +1205,81 @@
-                    e</var> to either resolve or reject.
-                  </li>
-                  <li>Unsubscribe <var>oldSubscription</var>.
-                  </li>
-                </ol>
+              <li>Dispatch <var>event</var> to <var>global</var>.
               </li>
             </ol>
           </li>
         </ol>
-        <p>
-          This algorithm ensures that the <a>Service Worker</a> is able to react to any
-          non-destructive change in a <a>push subscription</a>, such as an automatic refresh,
-          without causing any active <a>push subscription</a> to be terminated prematurely. A
-          <a>Service Worker</a> can request a new <a>push subscription</a> during this process and
-          ensure that no <a>push messages</a> are lost.
+        <p class="note">
+          Consider using a more reliable synchronization mechanism such as [[WEB-BACKGROUND-SYNC]]
+          when sending the details of the new <a>push subscription</a> to your
+          <a>application server</a>. The user might be subject to unreliable network conditions that
+          could cause a fetch to fail.
         </p>
+
+        <section data-dfn-for="PushSubscriptionChangeEvent">
+          <h2>
+            <dfn>PushSubscriptionChangeEvent</dfn> interface
+          </h2>
+          <pre class="idl">
+            dictionary PushSubscriptionChangeInit : ExtendableEventInit {
+              PushSubscription newSubscription = null;
+              PushSubscription oldSubscription = null;
+            };
+
+            [Constructor(DOMString type, optional PushSubscriptionChangeInit eventInitDict), Exposed=ServiceWorker]
+            interface PushSubscriptionChangeEvent : ExtendableEvent {
+              readonly attribute PushSubscription? newSubscription;
+              readonly attribute PushSubscription? oldSubscription;
+            };
+          </pre>
+          <p>
+            The <dfn>newSubscription</dfn> attribute contains the details of the
+            <a>push subscription</a> that is valid per invocation of the <a>pushsubscriptionchange</a>
+            event. The value will be <code>null</code> when no new <a>push subscription</a> could be
+            established, for example because the <a>webapp</a> has lost <a>express permission</a>.
+          </p>
+          <p>
+            The <dfn>oldSubscription</dfn> attribute contains the details of the
+            <a>push subscription</a> that SHOULD NOT be used anymore. The value will be
+            <code>null</code> when the <a>user agent</a> is not able to provide the full set of
+            details, for example because of partial database corruption.
+          </p>
+          <p class="issue">
+            Should we include an attribute that indicates whether the <code>oldSubscription</code>
+            has been decisively invalidated? How would developers use this information?
```

Comment:
I agree. Allowing the old subscription to briefly receive messages is helpful, but I don't think we want folks depending on it.